lib/sles4sap/sap_deployment_automation_framework/deployment.pm

```diff
@@ -562,8 +563,28 @@ sub prepare_sdaf_project {
     assert_script_run("cd $deployment_dir");
     assert_script_run('mkdir -p ' . log_dir());
 
+    # Calculate SDAF version used for deployment and picks latest -1
+    # SDAF_GIT_AUTOMATION_BRANCH variable will override calculated value
+    my $branch = get_var('SDAF_GIT_AUTOMATION_BRANCH');
+    if (!$branch || $branch eq 'latest') {
+        my $tags = script_output("curl -s https://github.com/Azure/sap-automation/tags | grep /releases/tag/ | grep data-view-component | cut -d '/' -f6 | cut -d '\"' -f1");
+        record_info("Releases: $tags");
+        my @releases = split('\n', $tags);
+        my $branch_expected = $releases[1];
+        if ($branch eq 'latest') {
```

Comment:
If there is no `SDAF_GIT_AUTOMATION_BRANCH` setting, `$branch` will be undef here, and there will be a warning for `$branch eq 'latest'`.

Reply:
Where is the warning I checked https://openqaworker15.qa.suse.cz/tests/304778#downloads and found nothing.

Reply:
Do you mean `Use of uninitialized value $branch in string eq at /var/lib/openqa/pool/49/os-autoinst-distri-opensuse/lib/sles4sap/sap_deployment_automation_framework/deployment.pm line 574.`?

Reply:
Fixed via `my $branch = get_var('SDAF_GIT_AUTOMATION_BRANCH', '');`

Reply:
New VR: https://openqaworker15.qa.suse.cz/tests/304843/logfile?filename=autoinst-log.txt (logs don't have `Use of uninitialized value $branch` any more).